### PR TITLE
Forensics Fixes

### PR DESCRIPTION
**Pull request by NanakoAC**
A package of bugfixes and tweaks for forensic techs/detectives.

---------------------------
Originally i set out to fix the longstanding issue of undesireable forensic interaction.
When you swab blood, you also whack the floor with the swab. Same with taking fingerprints, fibers, etc. Using these tools on various machines and objects caused other odd behaviours, and some behaviours while desireable (like putting a swab away in a box) made evidence collecting from such atoms impossible.

I tested a number of simpler solutions to fix this, but ran into various problems, and ended up going for the following
To fix this, i've tweaked the three main sampling tools : Swabs, fiber collector and fingerprint powder, to be state machines. attackself them to toggle their state between Storage mode and evidence collection.

When in storage mode, they can't collect evidence, and will interact normally with anything. 
In evidence collection mode, they intercept the click and make sure they ONLY attempt to collect evidence. No contaminating the target with your own prints, no whacking it, no triggering other wierd behaviours, and no inserting the swab into the target atom.

Powder/fiber kits have to be toggled back to storage mode before you can put them away. Swab kits, since they're single use, will automatically switch to storage mode after collecting something. (but they can still be manually switched, incase you take one out, don't collect anything and want to put it away again)

This has the result of making these tools a little more fiddly to use, but it allows the player to always do what they intended, and allows things that were previously impossible, like collecting blood/prints from storage containers, and from the outside of forensic machines. 
 And in a meticulous job like forensics, a little mechanical complexity isn't a bad thing.

Secondly, while looking through fingerprint code, i found an oddity. 
All gloves except latex entirely prevent leaving fingerprints. And latex gloves have a 25% chance of leaving fingerprints.

This seemed nonsensical both from a realism and gameplay standpoint. As latex gloves are only found in a few places, but everyone can get normal gloves. Soundscopes mentioned this isn't how things used to be, and it may have been a logic error that slipped in at some point.  And as can be seen from a quick image search, latex/nitrile gloves are the choice of forensic professionals to prevent evidence contamination https://duckduckgo.com/?q=forensic+technician&t=ffab&iax=1&ia=images

So i've inverted that behaviour. Now, latex gloves will never leave prints, and all other gloves have a small chance to leave prints, nerfed it to 10% rather than 25%, so wearing gloves is still helpful when committing a crime, but not perfectly reliable unless you go to the effort of stealing latex gloves from medical or forensics.

Added a box of latex gloves to the crime scene kit, in light of the above change, and made the forensic tech start off wearing them

Enabled weight based storage on the crime kit, i finally figured out how. You set the storage slots of the container to null, and and it activates the proportional storage.

Also enabled it on the BST backpack

### CODE REVIEW

**Comment by NanakoAC** [comment]
Should probably target development, since this got a little more complex than expected. I cant seem to find the button to retarget

**Comment by Lohikar** [comment]
Looks like master'll need merging into development first.

File: code/modules/detectivework/tools/swabs.dm [hunk 1]
```diff
@@ -126,3 +139,4 @@
 	desc = "[initial(desc)] The label on the vial reads 'Sample of [sample_str] from [source].'."
 	icon_state = "swab_used"
 	used = 1
+	collecting_evidence = 0
```
Comment (skull132) [inline code/modules/detectivework/tools/swabs.dm]: I kinda want to pester you to change the name. "Collecting evidence" implies that the variable is used to determine whether or not the user is actively collecting evidence. (In the process of doing so at this very moment.) Where as the actual semantics are more in line with "accepting evidence".

File: code/modules/detectivework/tools/swabs.dm [hunk 2]
```diff
@@ -75,6 +83,11 @@
 		user << "<span class='warning'>This swab has already been used.</span>"
 		return
 
+	if (!collecting_evidence)
```
Comment (skull132) [inline code/modules/detectivework/tools/swabs.dm]: Checking the same thing twice. For some raisin.

**Comment by LordFowl** [comment]
From what I recollect, in old code black gloves would entirely block prints, regular gloves would leave partial prints, and latex would leave total prints. I believe this to be the optimal situation - latex gloves are far more common than black gloves, especially to medicians.

File: code/modules/detectivework/tools/swabs.dm (hunk 2 above)
Comment (NanakoAC) [inline code/modules/detectivework/tools/swabs.dm]: oops, fixing

**Comment by NanakoAC** [comment]
sterile rubberlike gloves are the choice of real forensic experts for a reason,. the material is nonpermeable and prevents contamination of evidence, as well as being disposable so you can toss them away after they get covered in blood or somesuch. They should not be leaving prints, and are the most thematically appropriate choice for forensics work

Being available to medical is a perk of the job for traitors. Asking for some from those places that have them is slightly suspicious, since they aren't exactly a fashion choice. They are similarly noticeable and odd looking when worn

Other gloves, particularly those made of fabrics, can wear and parts of your skin can poke through. Not to mention being widely available in custom loadouts.
I don't see any reason why a specific color should affect this, an exemption for black gloves seems nonsensical

Aside from that, game balance is a thing to worry about, With the existing behaviour there's hardly any prints everywhere because near everyone gets some kind of gloves from their job, and is wearing them constantly. Latex gloves are available to a more limited subset of people whose work deals with sterility. And also to cargo, who have to order them

**Comment by NanakoAC** [comment]
skull's requested changes done

**Comment by LordFowl** [comment]
I remain opposed to it - realism is an unnecessary consideration because the forensics system in itself is not particularly realistic in terms of depth, thus fingerprints are the cornerstone of its operations. It is fact that there are far more latex gloves than there are black gloves, especially with suggestions that border them closer and closer to becoming one-use disposable, easily replenishable items. Meanwhile black gloves are only available to a handful of jobs. Black gloves are also, obviously, different from other gloves. Like insulated gloves, as indicated by their other mechanical features, such as being the only glove-type safely capable of removing lightbulbs. They are thicker, sturdier, and more impermeable than other gloves.

You also seem to have entirely discounted the more pertinent point of my objection, in that in the old system non-black, non-latex gloves left _partial_ prints, which actually added a further level of deduction and possibility of error to the FT's work.

**Comment by NanakoAC** [comment]
Believability, not realism, is the goal. People expect latex gloves to not leave prints, ive seen at least one antag caught because of this entirely reasonable belief being incorrect

Black gloves are also a common fashion accessory and entirely normal to wear, which removes all thought from the process.  The lightbulb thing is really minor, i'm not sure what to make of that, but black gloves are very common, They're available in custom loadout and already desired for looks. Hence there's no suspicion inherent in wearing them, regardless of your job. I also don't see the merit in adding more value to something that's already common and desireable, it will just farther reduce the variety of gloves worn.

Aquiring latex gloves may not be especially difficult but they are not commonly worn and doing so marks you out. The people who have IC reason to wear them are also (or should be) regularly disposing and replacing them

As for partial prints, i'm not sure how those work, the main purpose here was correcting the problem of latex leaving prints.  Leaving partial prints in those situations sounds interesting. I agree with this feature, though its beyond the scope of what i set out to do today

**Comment by NanakoAC** [comment]
looking closer, it seems that partial prints are already implemented, though gloves aren't factored in.

From what i can tell, the way it currently works is that EVERY print is always partial, but repeated prints on the same object get clearer and clearer until it eventually becomes full.

**Comment by LordFowl** [comment]
This is not "adding more value to an item that is already desirable", this is clearly removing value from an item. The added prevalancy of black gloves, particularly to loadouts, is something we in the business call "power creep". I would be all for boosting the price or altogether removing black gloves from the loadout, as they are functionally and canonically distinct from other gloves.

I agree that there is a logical error in everything being printproof, and if partial prints no longer exist (which clearly needs to be tested, as you seem to only have examined this code-wise, not function-wise), I would agree to making non black-gloves leave prints. My concensus ends there, however.

Furthermore, black gloves _are_ visibly distinct from regular gloves. Compare "These work gloves are thick and fire-resistant." to "A pair of gloves, they don't look special in any way.".

Finally, you seem to have made a radical assumption based off of pictures of forensic technicians you found on the internet. Latex gloves do not magically prevent leaving fingerprints - in fact such thin gloves are, according to my brief research, _more_ likely to leave prints than regular gloves. Even if your natural finger oils do not permeate through the glove, which is something of an if, form-fitting latex gloves _will_ preserve the ridges of your fingerprint, meaning that they will still leave such a print on any oily surface - re; pretty much any object on a frequently inhabited space station, or if the gloves themselves become oily, for example if you were to touch any other part of your body. Forensic technicians wear latex gloves to prevent DNA contamination, not fingerprint contamination, which is not something that is reflected in our code, and even so they have to switch off and dispose their gloves pretty much constantly. With even some rudimentary research it becomes clear that this claim for realism or believability is somewhat miscarried.

**Comment by NanakoAC** [comment]
To give black gloves another exclusive ability would definitely be adding power to them. The fact that they currently have it is not as relevant as it being taken away from other things.

Putting latex gloves below other gloves in forensic terms is just... i cant think of a word
Google images was merely used as an example, i could go dig up hundreds of others,. Thin rubber gloves ARE used for forensics, for a variety of reasons, they are a signature item.

I concede that there is potential for fingerprint transfer through them, but only when they have oils on the surface, and most likely also when the hand inside them has sweated enough for the material to tightly adhere to the hand. Putting on a fresh pair is generally sound for preventing evidence contamination. 

DNA contamination is a mechanic we don't have at the moment, yes. I'm not sure what to do about that right now, but it is a major reason for the wearing of such gloves in this field. We have a problem of mechanical bias here, and your proposal will result in forensic techs wearing bulky work gloves in the lab purely for mechanical reasons. 

There are several other mitigating mechanics we're missing too, like glove prints: Work gloves tend to have a unique textured rubber coating which is as unique as a fingerprint, and leather gloves, being organically derived, too carry a unique pattern of wrinkles from the skin of the animal. There's also smudging: Thin gloves are worn in many professions - including by thieves, surgeons and forensic techs, for their minimal effect on manual dexterity which allows more precise movements, like carefully handling an object covered in evidence without destroying said evidence. Someone in this field is going to be trained in careful handling to avoid contamination or destruction. We do not have any mechanical implementation of skills to represent that either

I did say believability, not realism. I'm pushing for latex gloves on forensic techs for roleplay reasons. its a thing that they do. Believable roleplay in forensics should involve wearing the appropriate gear. But right now we have mechanical disincentives to do that, and your proposal is to actually make it worse. This is a bigger picture deal.
If we try to go for completely accurate simulation of the details, our bigger picture looks fucked up, because we're missing several of those details. We have to work with what we've got, and i'm trying to alter mechanics to reward and incentivise towards realistic, RP-appropriate behaviour

It looks like we have a controversial issue here, maybe we can find a compromise?

**Comment by LordFowl** [comment]
>we need to go for realism and believability!
>btw, this PR is justified because blocking lasers with an e-sword is unrealistic.

My primary concern remains just as it was - latex gloves are more common than black gloves ought to be. Black gloves occupy a similar tier as insulated gloves, and should be restricted as such. Yes, black gloves are better than regular gloves. Good. The issue now is that they for some reason are equally available via loadouts. Bad.

Bad. All gloves should leave partial prints, latex gloves should just leave more. Isn't that what you accomplished in this PR, except a little somewhat in reverse?

Perhaps they are a signature item, but they are also a signature item for doctors, and there are a lot more doctors than there are forensic technicians. It makes the FT's job unduly difficult for roughly a quarter of the staff to go around leaving no fingerprints.

For an issue of realism as minor as gloves, I think mechanical bias should prevail over realism. It is not terribly immersion breaking for CSI to be wearing thicker, impermeable gloves, because it's really inconsequential. The mechanical repercussions of giving latex gloves a similar ability are not so inconsequential, as I've highlighted above.

The **only** compromise I can see is removing gloves being impermeable completely - fingerprints can be left regardless of what glove you're wearing, except for perhaps very thick gloves like insulated gloves or combat gloves. This would be the most realistic change, and this change really should not affect the FT much, because the rest of your changes here ensure that they cannot contaminate a scene by using their own tools, rendering gloves largely unnecessary because they will be able to collect prints without leaving their own. Beyond that, it'd promote better crime scene caution - a lack thereof being the most unbelievable trait a FT could display.

**Comment by NanakoAC** [comment]
How about this:
I make latex gloves fully block prints, but only for a short time after being first put on. say 10 minutes, after that they're contaminated, sticking to your hands, and will leave more prints than other gloves. 
Someone interested in using them to do crime without leaving prints would need to steal a whole box, rather than just a pair, and they'd have to regularly dispose of them.


Also worth noting (current behaviour), while wearing gloves, you not only leave prints on the environment, but you also leave prints on (presumably the inside of) your own gloves. So if you dispose of them and they're found, they can be tested and it thus proven that you wore them

In addition, RE the black gloves, i'd say we rename them leather gloves, give them the full print immunity and raise their loadout cost to 2 points (optionally also add a non-special pair of black gloves just for looks, with the normal cost)

How does this sound for compromise?

**Comment by NanakoAC** [comment]
@LordFowl

**Comment by LordFowl** [comment]
I am not terribly well-inclined towards that solution, although it is far superior to the initial premise of this PR. I am actually best inclined towards the alternative I presented, even above the idea of keeping everything the same as it is on live.

Do note however that we already have leather gloves, used botanically.

File: code/modules/detectivework/tools/sample_kits.dm [hunk 3]
```diff
@@ -128,8 +128,20 @@
...
+	..()
+	user << "<span class='notice'>It is in [accepting_evidence ? "evidence collection" : "storage"] mode.</span>"
+
+/obj/item/weapon/forensics/sample_kit/attack_self(var/mob/user)
+	accepting_evidence = !accepting_evidence
+	user << "<span class='notice'>[src] is now in [accepting_evidence ? "evidence collection" : "storage"] mode.</span>"
 
 /obj/item/weapon/forensics/sample_kit/proc/can_take_sample(var/mob/user, var/atom/supplied)
+	if (!accepting_evidence)
+		user << span("warning","[src] is in storage mode and not taking evidence. Click it to toggle modes.")
```
Comment (Lohikar) [inline code/modules/detectivework/tools/sample_kits.dm]: Don't suppose this message could be converted to IC, could it?

File: code/modules/detectivework/tools/sample_kits.dm [hunk 4]
```diff
@@ -128,8 +128,20 @@
...
 	var/evidence_path = /obj/item/weapon/sample/fibers
+	accepting_evidence = 0
+
+/obj/item/weapon/forensics/sample_kit/examine(var/mob/user)
+	..()
+	user << "<span class='notice'>It is in [accepting_evidence ? "evidence collection" : "storage"] mode.</span>"
+
+/obj/item/weapon/forensics/sample_kit/attack_self(var/mob/user)
+	accepting_evidence = !accepting_evidence
+	user << "<span class='notice'>[src] is now in [accepting_evidence ? "evidence collection" : "storage"] mode.</span>"
```
Comment (Lohikar) [inline code/modules/detectivework/tools/sample_kits.dm]: And this one.

File: code/modules/detectivework/tools/sample_kits.dm [hunk 5]
```diff
@@ -155,4 +167,7 @@
 	evidence_path = /obj/item/weapon/sample/print
 
 /obj/item/weapon/forensics/sample_kit/powder/can_take_sample(var/mob/user, var/atom/supplied)
+	if (!accepting_evidence)
+		user << span("warning","[src] is in storage mode and not taking evidence. Click it to toggle modes.")
```
Comment (Lohikar) [inline code/modules/detectivework/tools/sample_kits.dm]: And this one.

File: code/modules/detectivework/tools/swabs.dm [hunk 6]
```diff
@@ -66,6 +67,13 @@
 		return
 	return 1
 
+/obj/item/weapon/forensics/swab/attack_self(var/mob/user)
+	if (!used)
+		accepting_evidence = !accepting_evidence
+		user << "<span class='notice'>[src] is now in [accepting_evidence ? "evidence collection" : "storage"] mode.</span>"
```
Comment (Lohikar) [inline code/modules/detectivework/tools/swabs.dm]: And this one.

File: code/modules/detectivework/tools/swabs.dm [hunk 7]
```diff
@@ -75,6 +83,10 @@
 		user << "<span class='warning'>This swab has already been used.</span>"
 		return
 
+	if (!accepting_evidence)
+		user << "[src] is in storage mode and not taking evidence. Click it to toggle modes."
```
Comment (Lohikar) [inline code/modules/detectivework/tools/swabs.dm]: And this one.

File: code/game/jobs/job/security.dm [hunk 8]
```diff
@@ -132,9 +127,21 @@
 		if(H.mind.role_alt_title && H.mind.role_alt_title == "Forensic Technician")
 			H.equip_to_slot_or_del(new /obj/item/clothing/suit/storage/forensics/blue(H), slot_wear_suit)
 			H.equip_to_slot_or_del(new /obj/item/weapon/storage/briefcase/crimekit, slot_r_hand)
+			if(istajara(H))
+				H.equip_to_slot_or_del(new /obj/item/clothing/gloves/latex/tajara(H), slot_gloves)
+			if(isunathi(H))
```
Comment (Lohikar) [inline code/game/jobs/job/security.dm]: Doesn't look like this was your doing, but this should be an `else if`, not `if`.

**Pull request closed without merging**